templates/task-git-cli.yaml

```diff
@@ -0,0 +1,148 @@
+{{- end }}
+spec:
+  description: >-
+    This task can be used to perform git operations.
+
+    Git command that needs to be run can be passed as a script to
+    the task. This task needs authentication to git in order to push
+    after the git operation.
+
+  workspaces:
+    - name: source
+      description: A workspace that contains the fetched git repository.
+
+    - name: input
+      optional: true
+      description: |
+        An optional workspace that contains the files that need to be added to git. You can
+        access the workspace from your script using `$(workspaces.input.path)`, for instance:
+
+          cp $(workspaces.input.path)/file_that_i_want .
+          git add file_that_i_want
+          # etc
+
+    - name: ssh-directory
+      optional: true
+      description: |
+        A .ssh directory with private key, known_hosts, config, etc. Copied to
+        the user's home before git commands are executed. Used to authenticate
+        with the git remote when performing the clone. Binding a Secret to this
+        Workspace is strongly recommended over other volume types.
+
+    - name: basic-auth
+      optional: true
+      description: |
+        A Workspace containing a .gitconfig and .git-credentials file. These
+        will be copied to the user's home before any git commands are run. Any
+        other files in this Workspace are ignored. It is strongly recommended
+        to use ssh-directory over basic-auth whenever possible and to bind a
+        Secret to this Workspace over other volume types.
+
+  params:
+    - name: GIT_USER_NAME
+      type: string
+      description: |
+        Git user name for performing git operation.
+      default: ""
+
+    - name: GIT_USER_EMAIL
+      type: string
+      description: |
+        Git user email for performing git operation.
+      default: ""
+
+    - name: GIT_SCRIPT
+      description: The git script to run.
+      type: string
+      default: |
+        git help
+
+    - name: USER_HOME
+      description: |
+        Absolute path to the user's home directory. Set this explicitly if you are running the image as a non-root user or have overridden
+        the gitInitImage param with an image containing custom user configuration.
+      type: string
+      default: "/home/git"
+
+    - name: VERBOSE
+      description: Log the commands that are executed during `git-clone`'s operation.
+      type: string
+      default: "true"
+
+    - name: SSL_VERIFY
+      type: string
+      default: "true"
+      description: |
+        Sets the global `http.sslVerify` value, `false` is not advised unless
+        you trust the remote repository.
+
+    - name: DELETE_EXISTING
+      type: string
+      default: "true"
+      description: |
+        Clean out the contents of the `output` Workspace before cloning the
+        repository, if data exists.
+
+  results:
+    - name: COMMIT
+      description: The precise commit SHA after the git operation.
```

Comment:
Most of the things employed on this Task [is implemented on `task.yaml`](https://github.com/openshift-pipelines/task-git/blob/6c85da21eae3f29bd80869ff5d97006d7e0c254e/templates/task.yaml#L16-L132). I think we should follow the same design taken on `task-s2i.yaml` ([1](https://github.com/openshift-pipelines/task-containers/blob/3808765d18f17bdaf8a3a72d7df2cddfac2a45be/templates/spec-s2i.tpl) and [2](https://github.com/openshift-pipelines/task-containers/blob/3808765d18f17bdaf8a3a72d7df2cddfac2a45be/templates/task-s2i-go.yaml)), thus the user experience is the same for the Git related Tasks in this repository.